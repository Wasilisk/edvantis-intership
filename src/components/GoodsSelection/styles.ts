import styled from "styled-components";
import {ProductCardStyles} from "../Cards/ProductCard/styles";

export const GoodsSelectionStyles = styled.div`
  max-width: 1200px;
  margin: 0 auto;
  padding: 90px 24px;
  display: flex;
  justify-content: center;
  flex-direction: column;
  
  .text-info {
    text-align: center;
    
    h4 {
      font-size: 38px;
      color: #333333;
      line-height: 1.2;
      font-weight: 600;
      margin: 0;
    }
    
    p {
      font-size: 18px;
      line-height: 1.56;
      font-weight: 400;
      color: #333333;
      margin-top: 25px;
      margin-bottom: 0;
    }
  }
  
  .products-block {
    width: 100%;
    margin-top: 50px;
    display: flex;
    justify-content: center;
    & > ${ProductCardStyles} {
      flex-basis: 33.33333%;
      max-width: 33.33333%;
    }
  }

  @media (max-width: 768px) {
    .products-block {
      flex-direction: column;
      ${ProductCardStyles} {
        flex-basis: 100%;
        max-width: 100%;
      }
      
      ${ProductCardStyles}:not(:first-of-type) {
        margin-top: 50px;
      }
    }
  }
`