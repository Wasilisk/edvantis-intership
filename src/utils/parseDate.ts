const month = ["January","February","March","April","May","June","July","August","September","October","November","December"];

export const parseDate = (timestamp: number) => {
    const date = new Date(timestamp);

    return `${month[date.getMonth()]} ${('0' + date.getDate()).slice(-2)}, ${date.getFullYear()}`;
}