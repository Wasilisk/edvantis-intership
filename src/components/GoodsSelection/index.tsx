import React from 'react';
import { GoodsSelectionStyles } from './styles';
import {useAppSelector} from "../../hooks";
import {selectGoodsSelection} from "../../store/slices/HomePage/selectors";
import ProductCard from "../ProductCard";

const GoodsSelection = () => {
    const goodsSelectionData = useAppSelector(selectGoodsSelection);

    return (
        goodsSelectionData && <GoodsSelectionStyles>
            <div className="text-info">
                <h4>{goodsSelectionData.title}</h4>
                <p>{goodsSelectionData.description}</p>
            </div>
            <div className="products-block">
                {
                    goodsSelectionData.products.slice(0,3).map(product => {
                        return (
                            <ProductCard
                                key={product.id}
                                content={product}
                                buttonVariant="filled"
                            />
                        )
                    })
                }
            </div>
        </GoodsSelectionStyles>
    );
};

export default GoodsSelection;