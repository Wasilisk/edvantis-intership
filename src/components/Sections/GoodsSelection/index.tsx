import React from 'react';
import { GoodsSelectionStyles } from './styles';
import ProductCard from "../../Cards/ProductCard";
import {GoodsSelectionType} from "../../../models";

const GoodsSelection = ({title, description, products, id_name}: GoodsSelectionType) => {

    return (
        <GoodsSelectionStyles id={id_name}>
            <div className="text-info">
                <h4>{title}</h4>
                <p>{description}</p>
            </div>
            <div className="products-block">
                {
                    products.slice(0,3).map(product => {
                        return (
                            <ProductCard
                                key={product.id}
                                {...product}
                                buttonVariant="filled"
                            />
                        )
                    })
                }
            </div>
        </GoodsSelectionStyles>
    );
};

export default GoodsSelection;