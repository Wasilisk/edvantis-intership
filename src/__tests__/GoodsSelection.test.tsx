import {render} from "./utils/store-wrapper";
import GoodsSelection from "../components/GoodsSelection";
import {mockGoodsSelection} from "./__mock__/mock-goods-selection";
import {createMemoryHistory} from "history";
import {unstable_HistoryRouter as HistoryRouter} from "react-router-dom";

describe('GoodsSelection component', () => {
    it('should render component', () => {
        const history = createMemoryHistory({initialEntries: ['/']});
        const {getByText, asFragment, getAllByText} = render(
            <HistoryRouter history={history}>
                <GoodsSelection/>
            </HistoryRouter>, {
                initialState: {
                    home_page: {
                        goods_selection: mockGoodsSelection,
                        isLoading: false
                    }
                }
            });
        const sectionTitle = getByText(mockGoodsSelection.title);
        expect(sectionTitle).toBeDefined();

        const sectionDescription = getByText(mockGoodsSelection.description);
        expect(sectionDescription).toBeDefined();

        expect(getAllByText("Order Now").length).toBe(3);

        expect(asFragment()).toMatchSnapshot();
    });
})