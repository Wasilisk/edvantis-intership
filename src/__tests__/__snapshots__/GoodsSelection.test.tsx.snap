// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoodsSelection component should render component 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi johMkQ"
  >
    <div
      class="text-info"
    >
      <h4>
        Section title
      </h4>
      <p>
        Section description
      </p>
    </div>
    <div
      class="products-block"
    >
      <div
        class="sc-bczRLJ bRiLMA"
      >
        <a
          class="card-image"
          href="/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-dkzDqf hJRqwe"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ bRiLMA"
      >
        <a
          class="card-image"
          href="/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-dkzDqf hJRqwe"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ bRiLMA"
      >
        <a
          class="card-image"
          href="/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-dkzDqf hJRqwe"
        >
          Order Now
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
